File: hw/maisincho/u3/HW21TemplateMethod/TemplateMethodPy/model/Tea.py
from model.CaffeineBeverage import CaffeineBeverage

class Tea(CaffeineBeverage):
    def brew(self):
        print("Steep the tea")

    def add_condiments(self):
        print("Adding Lemon")

    def wants_condiments(self):
        answer = self.get_user_input()
        return answer.lower().startswith('y')

    def get_user_input(self):
        try:
            answer = input("Would you like lemon on your tea (y/n)? ")
        except Exception as e:
            print(f"Error reading input: {e}")
            answer = "no"
        return answer

File: hw/caizapanta/u3/HW21Template/src/main/java/ec/edu/espe/cafe/model/Tea.java
package ec.edu.espe.cafe.model;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * 
 * @author Tammy Caizapanta
 */

public class Tea extends CaffeineBeverage{
    
    @Override
    public void brew(){
        System.out.println("Steep the tea");
    }
    
    @Override
    public void addCondiments(){
        System.out.println("Adding lemon");
    }
    
    @Override
    public boolean wantsCondiments(){
        String answer = getUserInput();
        
        if(answer.toLowerCase().startsWith ("y")){
            return true;
        } else {
            return false;
        }
    }
    
    private String getUserInput(){
        String answer = null;
    System.out.println("Would you like lemon with your tea (y/n)?");

    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    
    try {
        answer = in.readLine();
    } catch (IOException e) {
        System.err.println("error when reading answer");
    }

    return answer;
    }
}